Issue 2  feat #1 회원 관리 기능

seyoung5744 opened the pull request:
변경사항
----
- #2 


AS-IS
-----
#### 회원가입
- 이메일, 이름, 비밀번호, ROLE을 통해 회원 가입.
- 이메일 중복이면 가입 불가.

#### 로그인 
- 이메일과 비밀번호를 통해 로그인.
- 로그인 성공 지JWT access Token 발행

#### 예외 처리 표준화
- 공통된 에러 응답을 위한 Global Exceprion을 구현.
- 에러 발생 시간, HTTP 상태 및 에러 코드, 설명을 응답.
- 에러 예시)

![에러 예시](https://user-images.githubusercontent.com/53173850/210754458-346179c8-a190-4950-9121-7cb540e52c89.png)

#### 구현된 테스트
- 회원가입 성공
- 이미 존재하는 이메일 일 때
- 로그인 성공



TO-BE
-----
- [ ] 이메일 인증
- [ ] 상품 데이터 크롤링 및 저장 구현 중
- [ ] 상품, 가격, 브랜드 별 CRUD

테스트 
-----
- [ ] 테스트 코드
- [ ] API 테스트

seyoung5744 committed:
feat : JWT 기반 회원 관리

회원 가입 시 이메일 중복 검사
로그인 시 이메일 유효 검사 후 토큰 발행

reference issue : [feat #1] 회원 관리 기능

seyoung5744 committed:
refactor : Contoller, Service 역할 분리

1. BaseEntity 분리
2. controller가 entity 직접 갖고 있지 못하도록 service로 추상화
3. token 반환을 String 타입에서 Token Class 타입으로

reference issue : [feat #1] 회원 관리 기능

seyoung5744 committed:
feat : 에러 처리 표준화

Custom Exception으로 Error 코드 표준화

Related to: #3 개발환경 구축

mingming-mentor commented on src/main/java/com/zerobase/musinsamonitor/controller/AuthController.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * 회원가입을 위한 API
+     */
+    @PostMapping("/signup")
```
signup으로 사용하셔도 좋지만 sign-up이나 signUp으로 사용하셔도 좋을 것 같아요 Restful API의 convention 규칙인데요, 아래 링크를 한번 참조해보시면 좋을 것 같습니다 https://prohannah.tistory.com/156

mingming-mentor commented on src/main/java/com/zerobase/musinsamonitor/security/jwt/TokenProvider.java:
```diff
@@ -0,0 +1,97 @@
+        return this.parseClaims(token).getSubject();
+    }
+
+    public boolean validateToken(String token){
```
validateToken이라는 네이밍은 token validation을 진행하면서 실패하면 exception을 던지는 형태의 네이밍으로 보여져요. t/f를 반환하고 싶다면 isValidToken등으로 is를 prefix로 붙여주시면 좋을 것 같습니다

mingming-mentor reviewed (approved):
안녕하세요. 밍밍멘토입니다. 코드 깔끔하게 잘 작성하셨네요. 간단히 몇가지 피드백 드렸습니다. 수고 많으셨어요!

seyoung5744 committed:
[refactor] Restful API 네이밍 규칙 적용

signup, signin과 같은 네이밍은 모호함
가독성을 위해 `-`(하이픈)을 사용하여 변경
signup => sign-up

seyoung5744 committed:
[refactor] Boolean타입 반환 is를 prefix 추가

validateToken이라는 네이밍은 token validation을 진행하면서 실패하면 exception을 던지는 형태의 네이밍으로 보여지므로 is를 prefix를 붙여서 가독성 향상
